profiles/common/modules/features/ec_embedded_image/ec_embedded_image.behat.inc

```diff
@@ -86,34 +86,21 @@ class ECEmbeddedImageSubContext extends DrupalSubContextBase {
       ]
     );
 
-    $this->contentTypeWasSetUp = TRUE;
+    $this->testPermissionsSetUp = TRUE;
   }
 
   /**
-   * Removes the test_media_gallery content type and related permissions.
+   * Removes the permissions related to the test_media_gallery content type.
+   *
+   * @see createMediaGalleryContentType()
    *
    * @AfterScenario
    */
-  public function removeMediaGalleryContentType() {
-    if (!$this->contentTypeWasSetUp) {
+  public function resetMediaGalleryPermission() {
+    if (!$this->testPermissionsSetUp) {
       return;
     }
 
-    $this->contentTypeWasSetUp = FALSE;
```

Comment:
I made some tests.
The class instance persists during the same Feature in different Scenarios.
The class instance does not persist for different Features.

So, I think we have a problem here. The roles that are currently being deleted after first scenario should remain until the last scenario in the feature is run.

Anyway, conclusions are:
# Class persist between scenarios in the same feature. I think `$this->contentTypeWasSetUp = FALSE;` should stay because it is out of scope to remove it and because by removing it we delete permissions twice.
# We should create a ticket to fix this problem of removing permissions right after first scenario.